titanic.py

```diff
@@ -93,12 +93,46 @@ def maximum_energy_delta(bqm):
                      for j in bqm.iter_neighbors(i))
                for i in bqm.iter_variables())
 
-def run_demo():
+def mutual_information_bqm(dataset, features):
+    """Build a QUBO that maximizes MI between survival and a subset of features"""
+    variables = ((feature, -mutual_information(prob(dataset[['survived', feature]].values), 1))
+                 for feature in features)
+    interactions = ((f0, f1, -conditional_mutual_information(prob(dataset[['survived', f0, f1]].values), 1, 2))
+                    for f0, f1 in itertools.permutations(features, 2))
+    return dimod.BinaryQuadraticModel(variables, interactions, 0, dimod.BINARY)
+
+def add_combination_penalty(bqm, k, penalty):
+    """Create a new BQM with an addition k-combination penalty"""
+    kbqm = dimod.generators.combinations(bqm.variables, k, strength=penalty)
+    kbqm.update(bqm)
+    return kbqm
+
+def mutual_information_feature_selection(dataset, features):
+    """Run the MIFS algoeith on D-Wave"""
+    
+    # Set up a QPU sampler with a fully-connected graph of all the variables
```

Comment:
This might confuse new users: at this point you haven't defined the graph. Perhaps something like, Set up a QPU sampler that embeds to a fully-connected graph of all the variables